src/components/Articles/ResolutionDates.vue

```diff
@@ -193,10 +195,26 @@ export default class ResolutionDates extends Mixins(CommonMixin) {
     }
   }
 
-  /** Updates store when resolution dates validity changes. */
-  @Watch('getIsResolutionDatesValid', { immediate: true })
-  private onEditingNameChanged (val: boolean): void {
-    this.setValidComponent({ key: 'isValidResolutionDate', value: val })
+  @Watch('hasRightsOrRestrictions')
+  @Watch('hasShareStructureChanged')
+  @Watch('isAdding')
+  private updateResolutionDatesValidity (): void {
+    if (this.isAdding) {
+      this.setValidComponent({ key: 'isValidResolutionDate', value: false })
+    } else {
+      this.setValidComponent({ key: 'isValidResolutionDate', value: !this.newResolutionDateRequired })
+    }
+  }
+
+  private get newResolutionDateRequired (): boolean {
```

Comment:
If it makes sense to you, move this (and maybe line 216) to the state-getters module (which is where some other such checks exist).
I'm not saying to change this now... I would say we don't have a complete design for such methods/getters atm so I leave it up to you.

Reply:
Oh, I now see you moved some of this out of state-getters. I'm sure you had a good reason. Ignore comment above and carry on :)

Reply:
I don't like the idea of adding a lot of logic to state-getters... actually it was in a state-getter and I refactored it to here.

Reply:
And, yes, I had a good reason. We need to discuss.

Reply:
I disagree, i think it makes sense for this getter to be global, as you are now pulling shareStructure information into this component.

I think using getters to compute some value is fine as long as it's specific and clear in its intent. 
My preference is to keep the components as simple as possible (i know it's tough sometimes), but moving this logic into the component only achieves 1 thing imo and that is making in accessible to anywhere else.

Reply:
As I said, we don't have a complete design for this :)

What do you (y'all) think of a mixin for methods with store logic?

Reply:
> I disagree, i think it makes sense for this getter to be global, as you are now pulling shareStructure information into this component.
Yes, I agree that this is not ideal dealing with shareStructure here.

> My preference is to keep the components as simple as possible (i know it's tough sometimes), but moving this logic into the component only achieves 1 thing imo and that is making in accessible to anywhere else.
I agree on keeping the components as simple as possible. I think we can find another approach other than using state-getters because when you do it, you are overriding the real state (the state will show is `isValidResolutionDate === true` but the getter may result in false for example)

Creating a mixin is a better alternative imo.

Reply:
> I think using getters to compute some value is fine as long as it's specific and clear in its intent.
My preference is to keep the components as simple as possible (i know it's tough sometimes), but moving this logic into the component only achieves 1 thing imo and that is making in accessible to anywhere else.

I strongly agree with this.

However, atm it's a guideline and not a strong design.

Reply:
I don't fully understand how a getter is overriding real* state, where the computed value is merely a representation of what the store values are. 

Mixin's are great for sure, but are purposed for used/extended by components. 
I think we've used getters to compute some values in the past because the easy access to store, aka `state`.

There is always a better way for sure, i'm 100% open to trying anything!

Reply:
Andre, why don't you try something that makes sense to you (looks/feels like good design), and we can review it and hopefully follow along with our later work?